### ISSUE TITLE
lastUpdated or lastModified field

### ISSUE THREAD

**Issue by barrycarter** [schemaorg/schemaorg#538]
Add a lastUpdated or lastModified (or updated or modified) field to all Things to indicate how recent/current the data you are providing is. Things (and Things) change over time, so have some sort of timestamp would be useful.

**Comment by danbri** [comment]
This might work better as properties of the WebPage or whatever that holds the description. In general properties attached to a Person object are considered to be of that real world entity. So a lastModified property of a Person e.g. me, would be saying something (slightly strange) about that Person, i.e. that the person was modified.

Does http://schema.org/dateModified on CreativeWork address your use case?

**Comment by mfhepp** [comment]
We should discuss whether it makes sense to add elements for protocol-level meta-data to schema.org.

The last modified data for a Web resource is typically returned in the HTTP response header when fetching the data.

There is an RDF vocabulary for HTTP protocol elements available at

    http://www.w3.org/TR/HTTP-in-RDF10/

I would hesitate to include such elements to schema.org. Search engine crawlers have to use a multiplicity of signals from the sites for assessing the validity and freshness of data anyway, so a simplistic declarative approach will not really help, IMO.

Martin

-----------------------------------
martin hepp  http://www.heppnetz.de
mhepp@computer.org          @mfhepp

**Comment by barrycarter** [comment]
Suppose I know that Bob lived in Chicago in 2005, but don't know where 
(or even if) he is living today. 
 
I think schema.org should let me represent this information somehow. 
 
@danbri You are correct in saying this is meta-data, not data about 
"Bob" himself. Unfortunately, dateModified on creativeWork doesn't 
really apply here, since Bob isn't really a creativeWork (except 
possibly in a metaphorical sense) 
 
@mfhepp I don't think using HTTP header data (or other "freshness" 
indicators) would work here. It's quite possible I'm writing about 
Bob's history today, so the webpage is new, but the data is old. 
 
Perhaps a Thing4D or ThingAtGivenTime or some combination of Thing and 
Timestamp is needed here. 
 
As a more realistic example, consider a map that shows the United 
States in 1776, then in 1800, then in 1865, and so on. Each individual 
date would be important.

**Comment by mfhepp** [comment]
To date, schema.org is mostly about first-order statements ("Peter is happy"). Higher-order statements (e.g. statements about statements - "The NY Times says that Peter was happy") are known to increase the complexity of data processing. In pure RDF, RDF reification and Named Graphs can be used to make higher order statements, but since they increase the computational effort of dealing with such data, they are not broadly used.

The closest we have in schema.org is http://schema.org/Role, see http://blog.schema.org/2014/06/introducing-role.html for details.

This is not exactly what you want, because you seem to propose a generic reification mechanism so that each and every statement could be constrained to a context, be it temporal information or the agent or data source making the statement.

I do not think that schema.org should go that route. Role was a difficult birth, is not widely used as of now, and has created lots of tough modeling problems we are only beginning to understand.

If you want to model such information, I would suggest to start with Named Graphs in RDF and try to apply schema.org properties to both the first-order statements inside a Named Graph and the Named Graph itself. One could argue that a Named Graph is a http://schema.org/CreativeWork. So you can attache all your contextual information using the properties available for http://schema.org/CreativeWork and put your main statements into the graph inside.

In a nutshell, if needed, it should be better handled outside of schema.org and instead in the underlying data model (e.g. RDF).

**Comment by danbri** [comment]
/cc @rvguha @vholland 

All I'll say for now is that this stuff gets complex very quickly :)

There might be something we could do that relates the describing document (i.e. all the facts it asserts) to a DateTime, along the lines that the description was an accurate description of the world at that point in time. For more fine-grained qualification, http://schema.org/Role is the best option we have without stepping outside of simple triples.

**Comment by vholland** [comment]
@barrycarter Do you have a use case? Chances are, there is a cleaner solution to the problem. For example, is this a DataCatalog or Dataset? Those types might benefit from this sort of information.

**Comment by vholland** [comment]
Is this still relevant given the changes outlined in issue #688?

**Comment by danbri** [comment]
While I don't think #688 covers everything from Barry's scenarios it addresses some of the concerns raised. As I'm pretty sure we are unlikely to get into hi-fidelity 4-dimensional modelling of all kinds of thing, I'm going to go ahead and close this issue for now. @barrycarter - thanks for raising this, and please jump in via #688 or continue posting here if you think there's something important missing.

**Comment by chaals** [comment]
On Tue, 15 Sep 2015 21:25:58 +0200, vholland <notifications@github.com>  
wrote:

>
> Is this still relevant given the changes outlined in issue #688?
>

Yeah, I think so. It might be reasonable to claim that a webpage is a  
datafeed, and so a statement (or even the page) is an item with  
dateModified etc.

After all, that's part of the point of so-called "DRY" - decorating  
content as metadata instead of having it hidden where people don't notice  
it rot.

But we would have to explain that is how we expect people to do things,  
because I suspect for a large majority it would seem counter-intuitive,  
even if they get it when you explain it.

cheers

-- 
Charles McCathie Nevile - web standards - CTO Office, Yandex
  chaals@yandex-team.ru - - - Find more at http://yandex.com
